kube/manifests/prometheus-gcs.yaml

```diff
@@ -17,6 +17,9 @@ spec:
       labels:
         app: prometheus-gcs
         thanos-peer: "true"
+      annotations:
+        prometheus.io/scrape: "true"
```

Comment:
So... as far as I know, this is not recommended anymore (especially port annotation). ):

The problem is that using these we immediately block multiple container monitoring within single pod - and especially here in Prom + sidecar pod, both of the containers are meant to be scraped.

Internally at Improbable, we use port namings to indicate what port should be scraped in our kubernetes scraping config (e.g port with suffix `m-` etc), however, I would love to hear more about other ways to solve this use case. But, again, as far as I am aware, annotations are not the ideal path for it. What do you think @alvaroaleman?

Reply:
Yeah it's true the annotations can not be used to scrape more than one port, but IMHO this is not an issue here because usually a prometheus is configured with job like this:

```
    - job_name: prometheus
      static_configs:
        - targets:
          - "127.0.0.1:9090"
```

I've added that one to the manifests as well.

True the annotations can not be used to scrape more than one port but aside of that they are the best working solution I know of. They are also still part of both [the official Kubernetes example](https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml) and the sample Prometheus config that is in this repository.

Reply:
Ok, it works maybe for Prometheus + sidecar but fails for any other multicontainer pod. 

Anyway the point of this example is to help users to run thanos smoothly, not to dictate what is the official and best solution for scraping multiple containers in k8s... so I don't mind merging this, with  some comment that this does not allow multiple port scrape. 

I wouldn't follow prometheus example for k8s sd blindlessly, it was probably written year ago or 2 ago (annotation part) New kubernetes SD list targets per each container port, so annotations are just not that useful, since they are per pod. Any other votes on that? (: But yea, I am fine for adding comment and putting this as something that "works" for most people.